mathesar/serializers.py

```diff
@@ -54,11 +54,47 @@ class SimpleColumnSerializer(serializers.Serializer):
 
 
 class ColumnSerializer(SimpleColumnSerializer):
-    index = serializers.IntegerField(source='column_index', read_only=True)
+    name = serializers.CharField(required=False)
+
+    # Creation fields
+    type = serializers.CharField(source='plain_type', required=False)
     nullable = serializers.BooleanField(default=True)
     primary_key = serializers.BooleanField(default=False)
+
+    # Duplication fields
+    duplicate_column = serializers.IntegerField(required=False, write_only=True)
+    copy_data = serializers.BooleanField(default=True, write_only=True)
+    copy_constraints = serializers.BooleanField(default=True, write_only=True)
+
+    # Read only fields
+    index = serializers.IntegerField(source='column_index', read_only=True)
     valid_target_types = serializers.ListField(read_only=True)
 
+    def validate(self, data):
```

Comment:
Sorry, not sure I understand this one. We need to check if both or neither set of arguments exist here, doesn't that mean we can't modularize?

Reply:
@eito-fis I was thinking something like

```
def validate(self, data):
  self._validate_from_scratch_column(data)
  self._validate_duplicated_column(data)
  return data
```

I'm not sure if it would be worth it though, you might have too much repeated code.

Reply:
I think my confusion is more from that it seems like our conditions are dependent on both `from_scratch` and `duplicated` and therefore seems hard to separate? Ex: the first condition ensures there are only fields specific to a single path, which requires looking at the arguments for both paths at the same time.

Reply:
Yeah, that's why I think there would be too much repeated code between the two and it might not be worth it. Feel free to leave it as-is.